src/tests/functional/plugin/shared/src/snippets/mlp.cpp

```diff
@@ -0,0 +1,118 @@
+        ov::test::utils::InputGenerateData in_data;
+        // To avoid big relative errors in the vicinity of zero, only positive values are generated for bf16 precision
+        in_data.start_from = model_input.get_element_type() == ov::element::bf16 ? 0 : -1;
+        in_data.range = 2;
+        in_data.resolution = 256;
+        tensor =
+            ov::test::utils::create_and_fill_tensor(model_input.get_element_type(), targetInputStaticShapes[i], in_data);
+        inputs.insert({model_input.get_node_shared_ptr(), tensor});
+    }
+}
+
+void MLPBase::SetUp() {
+    std::vector<InputShape> input_shapes;
+    ov::element::Type prc;
+    ov::AnyMap additional_config;
+    init_params(input_shapes, prc, additional_config);
+    init_input_shapes(input_shapes);
+
+    const auto subgraph_model = get_subgraph();
+    function = subgraph_model->getOriginal();
+
+    configuration.insert(additional_config.begin(), additional_config.end());
+    if (!configuration.count("SNIPPETS_MODE")) {
+        configuration.insert({"SNIPPETS_MODE", "IGNORE_CALLBACK"});
+    }
+
+    inType = outType = prc;
+    setInferenceType(prc);
+    init_thresholds();
+}
+
+void MLPBase::init_thresholds() {
+}
+
+std::string MLP::getTestCaseName(testing::TestParamInfo<ov::test::snippets::MLPParams> obj) {
+    std::vector<InputShape> input_shapes;
+    std::vector<ov::element::Type> elem_types;
+    ov::element::Type prc;
+    size_t thread_count;
+    std::string target_device;
+    size_t num_nodes, num_subgraphs;
+    ov::AnyMap additional_config;
+    std::tie(input_shapes,
+             elem_types,
+             prc,
+             thread_count,
+             num_nodes,
+             num_subgraphs,
+             target_device,
+             additional_config) = obj.param;
+
+    std::ostringstream result;
+    for (size_t i = 0; i < input_shapes.size(); i++)
+        result << "IS[" << i << "]=" << input_shapes[i] << "_";
+    for (size_t i = 0; i < elem_types.size(); i++)
+        result << "T[" << i << "]=" << elem_types[i] << "_";
+    result << "ThreadNum=" << thread_count << "_";
+    result << "PRC=" << prc << "_";
+    result << "#N=" << num_nodes << "_";
+    result << "#S=" << num_subgraphs << "_";
+    result << "targetDevice=" << target_device << "_";
+
+    if (!additional_config.empty()) {
+        result << "_PluginConf";
+        for (auto& item : additional_config) {
+            result << "_" << item.first << "=" << item.second.as<std::string>();
+        }
+    }
+    return result.str();
+}
+
+void MLP::init_params(std::vector<InputShape>& input_shapes, ov::element::Type& prc, ov::AnyMap& additional_config) {
+    std::tie(input_shapes, m_input_types, prc, m_thread_count, ref_num_nodes, ref_num_subgraphs,
+             targetDevice, additional_config) = this->GetParam();
+}
+
+std::shared_ptr<SnippetsFunctionBase> MLP::get_subgraph() const {
+    return std::make_shared<ov::test::snippets::MLPSeqFunction>(inputDynamicShapes, m_input_types, 2);
```

Comment:
2 - definitely should be parameter which can be selected in tests

Reply:
Parametrized and extended test cases